cuda_core/cuda/core/experimental/_module.py

```diff
@@ -397,3 +400,12 @@ def handle(self):
             handle, call ``int(ObjectCode.handle)``.
         """
         return self._handle
+
+
+def _rebuild_objectcode_instance(
+    module: Union[bytes, str],
+    code_type: str,
+    symbol_mapping: Optional[dict] = None,
+) -> ObjectCode:
+    """Rebuild an ObjectCode instance from its serialized form."""
+    return ObjectCode._init(module, code_type, symbol_mapping=symbol_mapping)
```

Comment:
For simplicity (readability), I'd move this one line into `__reduce__`.

Question: What if `_handle` of `self` (in `__reduce__`) `is not None`?

Reply:
I inlined this.

Reply:
Ultimately we can't safely serialize the handle itself. But I do think it's important to make sure that when we reinflate the class, starting it off with a `None` handle doesn't result in a broken state. 

My reading of the code is that as long as any method of `ObjectCode` that accesses `_handle` is decorated with `@precondition(_lazy_load_module)`, it should be safe to use. If so, it looks like a new `_handle` can be obtained in whatever context the `ObjectCode` is being deserialized in.

Btw, my last commit to `__reduce__` was wrong, will update in sec.